src/hotspot/share/jfr/recorder/checkpoint/types/jfrTypeSet.cpp

```diff
@@ -905,7 +990,10 @@ static size_t teardown() {
     ClearKlassAndMethods clear(_writer);
     _artifacts->iterate_klasses(clear);
     JfrTypeSet::clear();
```

Comment:
JfrTypeSet::clear() sets _clear_artifacts to true already. Maybe remove that call or remove the adjoining _clear_artifacts = true.

Reply:
fixed.